toponetx/classes/path_complex.py

```diff
@@ -0,0 +1,1149 @@
+"""Path complex."""
+from collections.abc import Hashable, Iterable, Iterator, Sequence
+from itertools import chain
+from typing import Any
+
+import networkx as nx
+import numpy as np
+import scipy as sp
+from hypernetx import Hypergraph
+from networkx.classes.reportviews import EdgeView, NodeView
+
+from toponetx.classes.combinatorial_complex import CombinatorialComplex
+from toponetx.classes.complex import Complex
+from toponetx.classes.path import Path
+from toponetx.classes.reportviews import PathView
+
+__all__ = ["PathComplex"]
+
+
+class PathComplex(Complex):
+    """A class representing a path complex based on simple paths as proposed in (https://arxiv.org/abs/2308.06838). The original path complex is defined in (https://arxiv.org/pdf/1207.2834.pdf).
```

Comment:
Please use proper references (Reference section) for this. See for example the modules in TopoModelX